### PR TITLE
chore(Dockerfile): Refactor image to use ubuntu-slim

### PR DESCRIPTION
**Pull request by krancour**

### CODE REVIEW

File: glide.yaml [hunk 1]
```diff
@@ -20,3 +20,5 @@ import:
 - package: github.com/arschles/kubeapp
   subpackages:
   - api
+- package: github.com/docker/docker
+  version: 2b27fe17a1b3fb8472fde96d768fa70996adf201
```
Comment (arschles) [inline glide.yaml]: why did this need to be set? if indeed we need it, can it be done in a separate PR?

**Comment by arschles** [comment]
Removed my `LGTM2` because of https://github.com/deis/workflow-manager/pull/57#discussion_r64300464

File: glide.yaml (hunk 1 above)
Comment (krancour) [inline glide.yaml]: Try a fresh glide install from what was there before... you'll find it fails because the sha its pinned to does not exist in docker/docker. I resolved this by pinning docker/docker to its previous value. Note I did put this in its own commit.

Comment (arschles) [inline glide.yaml]: @krancour here's the result of a fresh `glide install` on your branch, from a non-existent `vendor` dir:

```console
glide instalENG000656:workflow-manager aaronschlesinger$ glide install
[INFO] Downloading dependencies. Please wait...
[INFO] Fetching updates for github.com/gorilla/mux.
[INFO] Fetching updates for bitbucket.org/ww/goautoneg.
[INFO] Fetching updates for github.com/asaskevich/govalidator.
[INFO] Fetching updates for github.com/davecgh/go-spew.
[INFO] Fetching updates for github.com/google/cadvisor.
[INFO] Fetching updates for github.com/arschles/testsrv.
[INFO] Fetching updates for github.com/jchauncey/kubeclient.
[INFO] Fetching updates for github.com/gorilla/context.
[INFO] Fetching updates for github.com/arschles/kubeapp.
[INFO] Fetching updates for github.com/emicklei/go-restful.
[INFO] Fetching updates for github.com/beorn7/perks.
[INFO] Fetching updates for github.com/google/gofuzz.
[INFO] Fetching updates for github.com/docker/docker.
[INFO] Fetching updates for github.com/blang/semver.
[INFO] Fetching updates for github.com/docker/go-units.
[INFO] Fetching updates for github.com/golang/protobuf.
[INFO] Fetching updates for github.com/ghodss/yaml.
[INFO] Fetching updates for github.com/arschles/assert.
[INFO] Fetching updates for github.com/golang/glog.
[INFO] Fetching updates for github.com/go-swagger/go-swagger.
[INFO] Fetching updates for github.com/juju/ratelimit.
[INFO] Fetching updates for github.com/kelseyhightower/envconfig.
[INFO] Fetching updates for github.com/matttproud/golang_protobuf_extensions.
[INFO] Fetching updates for github.com/opencontainers/runc.
[INFO] Fetching updates for github.com/pborman/uuid.
[INFO] Fetching updates for github.com/prometheus/client_golang.
[INFO] Fetching updates for github.com/prometheus/client_model.
[INFO] Fetching updates for github.com/prometheus/common.
[INFO] Fetching updates for github.com/prometheus/procfs.
[INFO] Fetching updates for github.com/satori/go.uuid.
[INFO] Fetching updates for github.com/spf13/pflag.
[INFO] Fetching updates for github.com/ugorji/go.
[INFO] Fetching updates for golang.org/x/net.
[INFO] Fetching updates for gopkg.in/yaml.v2.
[INFO] Fetching updates for k8s.io/kubernetes.
[INFO] Fetching updates for speter.net/go/exp/math/dec/inf.
[INFO] Downloading dependencies. Please wait...
[INFO] Setting references.
[INFO] Setting version for github.com/arschles/kubeapp to d970a2ec475267698e3cd6412b4204520b8d653d.
[INFO] Setting version for github.com/emicklei/go-restful to 777bb3f19bcafe2575ffb2a3e46af92509ae9594.
[INFO] Setting version for github.com/davecgh/go-spew to 3e6e67c4dcea3ac2f25fd4731abc0e1deaf36216.
[INFO] Setting version for github.com/gorilla/mux to 9c19ed558d5df4da88e2ade9c8940d742aef0e7e.
[INFO] Setting version for github.com/gorilla/context to a8d44e7d8e4d532b6a27a02dd82abb31cc1b01bd.
[INFO] Setting version for github.com/asaskevich/govalidator to 9699ab6b38bee2e02cd3fe8b99ecf67665395c96.
[INFO] Setting version for github.com/google/cadvisor to 546a3771589bdb356777c646c6eca24914fdd48b.
[INFO] Setting version for github.com/beorn7/perks to b965b613227fddccbfffe13eae360ed3fa822f8d.
[INFO] Setting version for github.com/docker/go-units to 0bbddae09c5a5419a8c6dcdd7ff90da3d450393b.
[INFO] Setting version for github.com/blang/semver to 31b736133b98f26d5e078ec9eb591666edfd091f.
[INFO] Setting version for github.com/jchauncey/kubeclient to 9818037c585aa7c70c7c79125a900e966e22f03e.
[INFO] Setting version for github.com/arschles/assert to 5c22e2eba944d89ddd5c81b2c4175e64ee828503.
[INFO] Setting version for github.com/google/gofuzz to bbcb9da2d746f8bdbd6a936686a0a6067ada0ec5.
[INFO] Setting version for github.com/docker/docker to 2b27fe17a1b3fb8472fde96d768fa70996adf201.
[INFO] Setting version for github.com/golang/glog to 44145f04b68cf362d9c4df2182967c2275eaefed.
[INFO] Setting version for github.com/go-swagger/go-swagger to ff42df2bf231f6a12383b36332f8364254ea86d6.
[INFO] Setting version for github.com/ghodss/yaml to 73d445a93680fa1a78ae23a5839bad48f32ba1ee.
[INFO] Setting version for github.com/kelseyhightower/envconfig to 91921eb4cf999321cdbeebdba5a03555800d493b.
[INFO] Setting version for github.com/matttproud/golang_protobuf_extensions to fc2b8d3a73c4867e51861bbdd5ae3c1f0869dd6a.
[INFO] Setting version for github.com/golang/protobuf to b982704f8bb716bb608144408cff30e15fbde841.
[INFO] Setting version for github.com/juju/ratelimit to 77ed1c8a01217656d2080ad51981f6e99adaa177.
[INFO] Setting version for github.com/arschles/testsrv to 1ca51fb1a3b6e612411883cf1c9dbc58319b1d5b.
[INFO] Setting version for github.com/prometheus/client_golang to 3b78d7a77f51ccbc364d4bc170920153022cfd08.
[INFO] Setting version for github.com/ugorji/go to f4485b318aadd133842532f841dc205a8e339d74.
[INFO] Setting version for github.com/opencontainers/runc to 7ca2aa4873aea7cb4265b1726acb24b90d8726c6.
[INFO] Setting version for github.com/pborman/uuid to ca53cad383cad2479bbba7f7a1a05797ec1386e4.
[INFO] Setting version for github.com/spf13/pflag to 08b1a584251b5b62f458943640fc8ebd4d50aaa5.
[INFO] Setting version for github.com/prometheus/client_model to fa8ad6fec33561be4280a8f0514318c79d7f6cb6.
[INFO] Setting version for github.com/satori/go.uuid to f9ab0dce87d815821e221626b772e3475a0d2749.
[INFO] Setting version for github.com/prometheus/procfs to 490cc6eb5fa45bf8a8b7b73c8bc82a8160e8531d.
[INFO] Setting version for golang.org/x/net to e7da8edaa52631091740908acaf2c2d4c9b3ce90.
[INFO] Setting version for github.com/prometheus/common to ef7a9a5fb138aa5d3a19988537606226869a0390.
[INFO] Setting version for speter.net/go/exp/math/dec/inf to 46a40649338836f5d25521ce343379da79ef2184.
[INFO] Setting version for gopkg.in/yaml.v2 to a83829b6f1293c91addabc89d0571c246397bbf4.
[INFO] Setting version for k8s.io/kubernetes to 3d6adabd365ed5f81a2a3b843c63626c27e4034d.
[INFO] Setting version for bitbucket.org/ww/goautoneg to 75cd24fc2f2c.
```

Comment (arschles) [inline glide.yaml]: @krancour an update - without the version set, a `glide up` seems to work as well, from a non-existent `vendor` dir:

```console
ENG000656:workflow-manager aaronschlesinger$ glide up
[INFO] Downloading dependencies. Please wait...
[INFO] Fetching updates for github.com/gorilla/mux.
[INFO] Fetching updates for github.com/arschles/testsrv.
[INFO] Fetching updates for k8s.io/kubernetes.
[INFO] Fetching updates for github.com/arschles/assert.
[INFO] Fetching updates for github.com/kelseyhightower/envconfig.
[INFO] Fetching updates for github.com/satori/go.uuid.
[INFO] Fetching updates for speter.net/go/exp/math/dec/inf.
[INFO] Fetching updates for github.com/arschles/kubeapp.
[INFO] Fetching updates for github.com/go-swagger/go-swagger.
[INFO] Fetching updates for github.com/docker/docker.
[INFO] Setting version for github.com/go-swagger/go-swagger to 0.5.0.
[INFO] Detected semantic version. Setting version for k8s.io/kubernetes to v1.2.5-beta.0.
[INFO] Resolving imports
[INFO] Fetching github.com/gorilla/context into /Users/aaronschlesinger/gocode/src/github.com/deis/workflow-manager/vendor
[INFO] Found Godeps.json file.
[INFO] Fetching github.com/google/cadvisor/info/v1 into /Users/aaronschlesinger/gocode/src/github.com/deis/workflow-manager/vendor
[INFO] Setting version for github.com/google/cadvisor to 546a3771589bdb356777c646c6eca24914fdd48b.
[INFO] Fetching github.com/davecgh/go-spew/spew into /Users/aaronschlesinger/gocode/src/github.com/deis/workflow-manager/vendor
[INFO] Setting version for github.com/davecgh/go-spew to 3e6e67c4dcea3ac2f25fd4731abc0e1deaf36216.
[INFO] Fetching github.com/ugorji/go/codec into /Users/aaronschlesinger/gocode/src/github.com/deis/workflow-manager/vendor
[INFO] Setting version for github.com/ugorji/go to f4485b318aadd133842532f841dc205a8e339d74.
[INFO] Fetching golang.org/x/net/context into /Users/aaronschlesinger/gocode/src/github.com/deis/workflow-manager/vendor
[INFO] Setting version for golang.org/x/net to c2528b2dd8352441850638a8bb678c2ad056fd3e.
[INFO] Setting version for speter.net/go/exp/math/dec/inf to 42ca6cd68aa922bc3f32f1e056e61b65945d9ad7.
[INFO] Fetching github.com/golang/glog into /Users/aaronschlesinger/gocode/src/github.com/deis/workflow-manager/vendor
[INFO] Setting version for github.com/golang/glog to 44145f04b68cf362d9c4df2182967c2275eaefed.
[INFO] Found GB manifest file.
[INFO] Fetching github.com/asaskevich/govalidator into /Users/aaronschlesinger/gocode/src/github.com/deis/workflow-manager/vendor
[INFO] Setting version for github.com/asaskevich/govalidator to 9699ab6b38bee2e02cd3fe8b99ecf67665395c96.
[INFO] Fetching gopkg.in/yaml.v2 into /Users/aaronschlesinger/gocode/src/github.com/deis/workflow-manager/vendor
[INFO] Setting version for gopkg.in/yaml.v2 to d466437aa4adc35830964cffc5b5f262c63ddcb4.
[INFO] Found Godeps.json file.
[INFO] Fetching github.com/google/gofuzz into /Users/aaronschlesinger/gocode/src/github.com/deis/workflow-manager/vendor
[INFO] Setting version for github.com/google/gofuzz to bbcb9da2d746f8bdbd6a936686a0a6067ada0ec5.
[INFO] Fetching github.com/emicklei/go-restful/swagger into /Users/aaronschlesinger/gocode/src/github.com/deis/workflow-manager/vendor
[INFO] Setting version for github.com/emicklei/go-restful to 777bb3f19bcafe2575ffb2a3e46af92509ae9594.
[INFO] Fetching github.com/blang/semver into /Users/aaronschlesinger/gocode/src/github.com/deis/workflow-manager/vendor
[INFO] Setting version for github.com/blang/semver to 31b736133b98f26d5e078ec9eb591666edfd091f.
[INFO] Fetching github.com/prometheus/client_golang/prometheus into /Users/aaronschlesinger/gocode/src/github.com/deis/workflow-manager/vendor
[INFO] Setting version for github.com/prometheus/client_golang to 3b78d7a77f51ccbc364d4bc170920153022cfd08.
[INFO] Fetching github.com/spf13/pflag into /Users/aaronschlesinger/gocode/src/github.com/deis/workflow-manager/vendor
[INFO] Setting version for github.com/spf13/pflag to 08b1a584251b5b62f458943640fc8ebd4d50aaa5.
[INFO] Fetching github.com/juju/ratelimit into /Users/aaronschlesinger/gocode/src/github.com/deis/workflow-manager/vendor
[INFO] Setting version for github.com/juju/ratelimit to 77ed1c8a01217656d2080ad51981f6e99adaa177.
[INFO] Fetching github.com/opencontainers/runc/libcontainer/cgroups/fs into /Users/aaronschlesinger/gocode/src/github.com/deis/workflow-manager/vendor
[INFO] Setting version for github.com/opencontainers/runc to 7ca2aa4873aea7cb4265b1726acb24b90d8726c6.
[INFO] Fetching github.com/pborman/uuid into /Users/aaronschlesinger/gocode/src/github.com/deis/workflow-manager/vendor
[INFO] Setting version for github.com/pborman/uuid to ca53cad383cad2479bbba7f7a1a05797ec1386e4.
[INFO] Fetching github.com/beorn7/perks/quantile into /Users/aaronschlesinger/gocode/src/github.com/deis/workflow-manager/vendor
[INFO] Setting version for github.com/beorn7/perks to b965b613227fddccbfffe13eae360ed3fa822f8d.
[INFO] Fetching github.com/golang/protobuf/proto into /Users/aaronschlesinger/gocode/src/github.com/deis/workflow-manager/vendor
[INFO] Setting version for github.com/golang/protobuf to b982704f8bb716bb608144408cff30e15fbde841.
[INFO] Fetching github.com/prometheus/client_model/go into /Users/aaronschlesinger/gocode/src/github.com/deis/workflow-manager/vendor
[INFO] Setting version for github.com/prometheus/client_model to fa8ad6fec33561be4280a8f0514318c79d7f6cb6.
[INFO] Fetching github.com/prometheus/common/expfmt into /Users/aaronschlesinger/gocode/src/github.com/deis/workflow-manager/vendor
[INFO] Setting version for github.com/prometheus/common to ef7a9a5fb138aa5d3a19988537606226869a0390.
[INFO] Fetching github.com/prometheus/procfs into /Users/aaronschlesinger/gocode/src/github.com/deis/workflow-manager/vendor
[INFO] Setting version for github.com/prometheus/procfs to 490cc6eb5fa45bf8a8b7b73c8bc82a8160e8531d.
[INFO] Fetching github.com/ghodss/yaml into /Users/aaronschlesinger/gocode/src/github.com/deis/workflow-manager/vendor
[INFO] Setting version for github.com/ghodss/yaml to 73d445a93680fa1a78ae23a5839bad48f32ba1ee.
[INFO] Found Godeps.json file.
[INFO] Setting version for github.com/docker/docker to 0f5c9d301b9b1cca66b3ea0f9dec3b5317d3686d.
[INFO] Fetching bitbucket.org/ww/goautoneg into /Users/aaronschlesinger/gocode/src/github.com/deis/workflow-manager/vendor
[INFO] Setting version for bitbucket.org/ww/goautoneg to '75cd24fc2f2c2a2088577d12123ddee5f54e0675'.
[INFO] Fetching github.com/matttproud/golang_protobuf_extensions/pbutil into /Users/aaronschlesinger/gocode/src/github.com/deis/workflow-manager/vendor
[INFO] Setting version for github.com/matttproud/golang_protobuf_extensions to fc2b8d3a73c4867e51861bbdd5ae3c1f0869dd6a.
[INFO] Fetching github.com/docker/go-units into /Users/aaronschlesinger/gocode/src/github.com/deis/workflow-manager/vendor
[INFO] Setting version for github.com/docker/go-units to 0bbddae09c5a5419a8c6dcdd7ff90da3d450393b.
[INFO] Downloading dependencies. Please wait...
[INFO] Setting references for remaining imports
[INFO] Project relies on 35 dependencies.
```

Comment (krancour) [inline glide.yaml]: Yes... it was previously failing trying to set docker/docker to 0f5c9d301b9b1cca66b3ea0f9dec3b5317d3686d.

The interesting thing here...

You've met success by doing `glide up`.  `make bootstrap` normally does a `glide install`, which uses the precise versions called out in the `glide.lock`, vs. what `glide up` does-- which is to update to the latest version of everything (within whatever constraints are described in `glide.yaml`). So you've stepped outside the typical workflow, so to speak, that would be followed for someone who had just cloned this repo for the first time (as I had).  I think one way or another we've got to get that fixed.

Comment (arschles) [inline glide.yaml]: right, sorry I wasn't unclear. I'm suggesting just doing a `glide up` and checking in the new `glide.lock` that results, instead of pinning the version in the `glide.yaml`

Comment (krancour) [inline glide.yaml]: Got it. I can get on board with that.

**Comment by arschles** [comment]
Re-added my `LGTM2`, assuming https://github.com/deis/workflow-manager/pull/57#discussion_r64451804 will be done

**Comment by krancour** [comment]
@arschles addressed as we discussed. Thanks for the pre-emptive LGTM. I'll tag as pending CI.
